index.js

```diff
@@ -1,11 +1,30 @@
 const core = require('@actions/core')
 const AWS = require('aws-sdk')
 
-const getFindings = async (ECR, repository, tag) => {
+/**
+ * @typedef {{
+ *  critical: number,
+ *  high: number,
+ *  medium: number,
+ *  low: number,
+ *  informational: number,
+ *  undefined: number,
+ *  total: number }} IgnoredCounts
+ */
+
+/**
+ * @param {AWS.ECR} ECR
+ * @param {string} repository
+ * @param {string} tag
+ * @param {boolean} useMaxResults
+ * @returns {AWS.Request|AWS.AWSError|null} Results, Error or `null`.
+ */
+const getFindings = async (ECR, repository, tag, useMaxResults = false) => {
   return ECR.describeImageScanFindings({
```

Comment:
You mentioned that the max limit is 1000. Actually, the results will get paginated, so we could [iterate through until there are no more pages](https://advancedweb.hu/how-to-paginate-the-aws-js-sdk-using-async-generators/).

Supporting this would be a more correct solution. And I worry about the correctness of allowing users to specify an ignore list and only checking the first 1000 items. What if they ignore an item that gets later bumped out of the first 1000 results?

It might be better to either implement pagination support or fail due to "we don't support ignore_list for scans with >1k findings, disable your ignore list or submit a PR". I'm ambivalent about whether it's worth implementing paging. How common are >1k finding reports in the wild? 🤷

Reply:
Yeah, fair call re: pagination. I've seen the pagination but given **the findings are ordered by severity**, we decided to just go with the max available, 1k.

To give you an idea of our current scan:
```
Vulnerabilities found:
 16 Critical (16 ignored)
 76 High 
305 Medium 
283 Low 
354 Informational 
 18 Undefined
=================
1052 Total (16 ignored)
```

So >1k findings but we only fail on critical findings. While some of them get addressed at some point, we might want to ignore some false positives. So we wouldn't need to have 1k results. For our purposes, <500 would do. Chances are that in general, people might not care about anything below medium severity.

Reply:
That makes sense. If you're ignoring 1000 CVEs, you have bigger problems...

Reply:
Having said that, I have just added an extra method based on your link to `getAllFindings` 🕺 
See ef38dcaea8ad74ab52bc890c5953e26e947c3ce8

Reply:
🦸‍♂️